modules/charon-core/src/main/java/org/wso2/charon3/core/extensions/UserManager.java

```diff
@@ -1,143 +0,0 @@
-/*
- * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.wso2.charon3.core.extensions;
-
-import org.wso2.charon3.core.exceptions.BadRequestException;
-import org.wso2.charon3.core.exceptions.CharonException;
-import org.wso2.charon3.core.exceptions.ConflictException;
-import org.wso2.charon3.core.exceptions.NotFoundException;
-import org.wso2.charon3.core.exceptions.NotImplementedException;
-import org.wso2.charon3.core.objects.Group;
-import org.wso2.charon3.core.objects.User;
-import org.wso2.charon3.core.utils.codeutils.Node;
-import org.wso2.charon3.core.utils.codeutils.SearchRequest;
-
-import java.util.List;
-import java.util.Map;
-
-/**
- * This is the interface for usermanager extension.
- * An implementation can plugin their own user manager-(either LDAP based, DB based etc)
- * by implementing this interface and mentioning it in configuration.
- */
-public interface UserManager {
```

Comment:
This is a main interface in Charon repo. Removing public interfaces are not recommended.  There may be users who use this class. Changes should be done in a backward-compatible manner.